ompi/mca/pml/ob1/pml_ob1_sendreq.c

```diff
@@ -125,6 +125,46 @@ static int mca_pml_ob1_send_request_free(struct ompi_request_t** request)
 
 static int mca_pml_ob1_send_request_cancel(struct ompi_request_t* request, int complete)
 {
+#if OPAL_ENABLE_FT_MPI
+    ompi_communicator_t* comm = request->req_mpi_object.comm;
+
+    if( REQUEST_COMPLETE(request) ) { /* way to late to cancel this one */
+        return OMPI_SUCCESS;
+    }
+    /* If FT is enabled then we allow canceling requests with a dead process.
+     * However, we should not release the request as we don't know if there are
+     * no pending fragments. This generates a memory leak as the send requests
+     * will never be recovered. */
+    if( !ompi_comm_is_proc_active(comm, request->req_peer,
+                                  OMPI_COMM_IS_INTER(comm)) ) {
+        mca_pml_ob1_send_request_t* pml_req = (mca_pml_ob1_send_request_t*)request;
+        /**
+         * As now the PML is done with this request we have to force the pml_complete
+         * to true. Otherwise, the request will never be freed.
+         */
+        opal_output_verbose(10, ompi_ftmpi_output_handle,
+                                "Send_request_cancel: cancel granted for request %p because peer %d is dead\n",
+                                (void*)request, request->req_peer);
+        request->req_status._cancelled = true;
+#if 0
```

Comment:
This should be kept in a separate branch, if we don't have the answer to the question

Reply:
There's an assert just above in existing code that says it never happens.